js/packages/ufg-client/src/resources/fetch-resource.ts

```diff
@@ -141,6 +141,19 @@ function handleTunneledResource({
         }
       }
     },
+    afterResponse({response}) {
+      const headers: Record<string, string> = {}
+      for (const [name, value] of response.headers) {
+        if (name.startsWith('x-fetch-response-header')) {
+          headers[name.replace('x-fetch-response-header-', '')] = value
+        }
+      }
+      return new Response(response.body, {
+        status: Number(response.headers.get('x-fetch-status-code')),
+        statusText: response.headers.get('x-fetch-status-text')!,
+        headers,
+      })
```

Comment:
I prefer to use that syntax instead because how we work with the `Response` and `Request` classes is tricky.
```suggestion
      return {
        body: response.body,
        status: Number(response.headers.get('x-fetch-status-code')),
        statusText: response.headers.get('x-fetch-status-text')!,
        headers,
      }
```

Reply:
I don't mind accepting your suggestion. But what is tricky about it? I saw that if I return a response, then that is used: https://github.com/applitools/eyes.sdk.javascript1/blob/cf34ad1a5b3cba0b29b3509616b20a2b1313c62f/js/packages/req/src/req.ts#L216 what is the difference?

Reply:
It works only because req is bundled with node-fetch, maybe in the future we will be able to avoid it, the api I suggested was created to overcome this limitation, also this is the way that we use in all other places.

Reply:
Got it. So I will accept your suggestion. Thanks!